### PR TITLE
feat(types,clerk-js): Add types for TOTP support on the setup level

### PR DESCRIPTION
**Pull request by yourtallness**
## Type of change

- [ ] 🐛 Bug fix
- [X] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [X] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [X] `@clerk/types`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

Adds support for the `TOTP` resource & 3 related operations on the User (`generateTOTP`, `verifyTOTP`, `deleteTOTP`).

### CODE REVIEW

File: packages/clerk-js/src/core/resources/User.ts [hunk 1]
```diff
@@ -146,6 +154,40 @@ export class User extends BaseResource implements UserResource {
     return new ExternalAccount(json, this.path() + '/external_accounts');
   };
 
+  generateTOTP = async (): Promise<TOTPResource> => {
+    const json = (
+      await BaseResource._fetch<ExternalAccountJSON>({
+        path: '/me/totp',
```
Comment (SokratisVidros) [inline packages/clerk-js/src/core/resources/User.ts:160]: @yourtallness The API is a bit different from the [design doc](https://www.notion.so/clerkdev/TOTP-Time-based-One-Time-Password-46320e464d024d938cc44c79468122da). It's `/me/totp` vs `/me/totp_secret`. Is this intentional?

**Review by SokratisVidros — APPROVED** [review]

File: packages/clerk-js/src/core/resources/User.ts [hunk 2]
```diff
@@ -146,6 +154,40 @@ export class User extends BaseResource implements UserResource {
     return new ExternalAccount(json, this.path() + '/external_accounts');
   };
 
+  generateTOTP = async (): Promise<TOTPResource> => {
```
Comment (chanioxaris) [inline packages/clerk-js/src/core/resources/User.ts:157]: Should we align the new methods naming with the existing ones?
- generateTOTP -> `createTOTP()`
- deleteTOTP -> `disableTOTP()`

File: packages/clerk-js/src/core/resources/User.ts [hunk 3]
```diff
@@ -146,6 +154,40 @@ export class User extends BaseResource implements UserResource {
...
+    const json = (
+      await BaseResource._fetch<ExternalAccountJSON>({
```
Comment (chanioxaris) [inline packages/clerk-js/src/core/resources/User.ts:159]:
```suggestion
      await BaseResource._fetch<TOTPJSON>({
```

File: packages/clerk-js/src/core/resources/User.ts [hunk 4]
```diff
@@ -146,6 +154,40 @@ export class User extends BaseResource implements UserResource {
...
+        method: 'POST',
+      })
+    )?.response as unknown as TOTPJSON;
+
+    return new TOTP(json);
+  };
+
+  verifyTOTP = async (code: string): Promise<TOTPResource> => {
+    const json = (
+      await BaseResource._fetch<ExternalAccountJSON>({
```
Comment (chanioxaris) [inline packages/clerk-js/src/core/resources/User.ts:170]:
```suggestion
      await BaseResource._fetch<TOTPJSON>({
```

File: packages/clerk-js/src/core/resources/User.ts [hunk 5]
```diff
@@ -146,6 +154,40 @@ export class User extends BaseResource implements UserResource {
...
+        path: '/me/totp',
+        method: 'POST',
+      })
+    )?.response as unknown as TOTPJSON;
+
+    return new TOTP(json);
+  };
+
+  verifyTOTP = async (code: string): Promise<TOTPResource> => {
```
Comment (chanioxaris) [inline packages/clerk-js/src/core/resources/User.ts:168]: We should pass a params object here  to make it more future proof

```suggestion
  verifyTOTP = async (params: VerifyTOTPParams): Promise<TOTPResource> => {
```

File: packages/clerk-js/src/core/resources/User.ts (hunk 1 above)
Comment (yourtallness) [reply packages/clerk-js/src/core/resources/User.ts:148]: The ddoc has been updated with the new naming

File: packages/clerk-js/src/core/resources/User.ts (hunk 3 above)
Comment (yourtallness) [reply packages/clerk-js/src/core/resources/User.ts]: Fixed

File: packages/clerk-js/src/core/resources/User.ts (hunk 2 above)
Comment (yourtallness) [reply packages/clerk-js/src/core/resources/User.ts:145]: Sure

File: packages/clerk-js/src/core/resources/User.ts (hunk 5 above)
Comment (yourtallness) [reply packages/clerk-js/src/core/resources/User.ts:156]: Yep

File: packages/clerk-js/src/core/resources/User.ts (hunk 4 above)
Comment (yourtallness) [reply packages/clerk-js/src/core/resources/User.ts:159]: Fixed

**Pull request merged**